File: 卡码/代码随想录/1/06-二叉树/Java/binaryTree/src/main/java/com/kurumi/binarytree/_04_二叉搜索树属性/LeetCode_0538.java
//538. 把二叉搜索树转换为累加树
package com.kurumi.binarytree._04_二叉搜索树属性;

import com.kurumi.binarytree.TreeNode;

public class LeetCode_0538 {

}
class Solution {
  int pre=0;
  public void traversal(TreeNode root){
    if(root==null) return;
    traversal(root.right);
    root.val+=pre;
    pre=root.val;
    traversal(root.left);
  }
  public TreeNode convertBST(TreeNode root) {
    pre=0;
    traversal(root);
    return root;
  }
}

File: 卡码/代码随想录/Go/二叉树/04-二叉搜索树属性/LeetCode_0538.go
//538. 把二叉搜索树转换为累加树
package main

import (
	. "binaryTree/types"
)

func convertBST(root *TreeNode) *TreeNode {
	var pre=0
	var traversal func(root *TreeNode)
	traversal=func(root *TreeNode){
		if root==nil {
			return
		}
		traversal(root.Right)
		root.Val+=pre
		pre=root.Val
		traversal(root.Left)
	}
	traversal(root)
	return root
}
